### PR TITLE
Enable Point of Care for ANC form

### PR DESCRIPTION
**Pull request by semujju**
https://app.asana.com/0/0/1198490262486240/f

### CODE REVIEW

**Comment by slubwama** [comment]
@semujju  add ticket to your PR. also Rename your PR to reflect the ticket

**Comment by semujju** [comment]
> @semujju add ticket to your PR. also Rename your PR to reflect the ticket

Done

File: api/src/main/resources/metadata/Custom_Concepts.xml [hunk 1]
```diff
@@ -5265,6 +5346,92 @@
...
+    <concept_name  concept_name_id="161514"  concept_id="166177"  name="Length of menses"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-02 15:39:29"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="c2b9bfb9-1f9b-4c65-8573-b7b93dc5f443"  />
+    <concept_name  concept_name_id="161515"  concept_id="166178"  name="Heavy"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-02 15:40:33"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="6bc5b559-1d9b-4c89-b85d-763b04c598d0"  />
+    <concept_name  concept_name_id="161516"  concept_id="166179"  name="Amount of menses"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-02 15:43:49"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="8c0dda10-5e40-4ea7-9145-61e765acbf72"  />
+    <concept_name  concept_name_id="161517"  concept_id="166180"  name="Menstrual & Contraceptive History"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-02 15:44:20"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="4ac3b380-ef5a-4edc-8d45-be12ef5cb64a"  />
```
Comment (slubwama) [inline api/src/main/resources/metadata/Custom_Concepts.xml:5370]: The & alone will cause a failure in the import. Remember to convert them before you add them to an import file

**Review by slubwama — CHANGES_REQUESTED** [review]

File: omod/src/main/webapp/resources/htmlforms/HMIS-005-IntegratedAntenatalRegister.xml [hunk 2]
```diff
@@ -307,33 +987,220 @@
...
+                                                    <obsgroup groupingConceptId="166239">
+                                            <div class="row obstetric" id="{id}-toggleContainer7">
+                                            <div class="col-md-9">
+
+                                            <div class="row">
+                                                <div class="col-md-2">
+                                                    <obs conceptId="166185" labelText="Pregnancy No." />
+                                                </div>
+                                                <div class="col-md-2 pt-4">
+                                                    <obs conceptId="50" style="checkbox" />
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/HMIS-005-IntegratedAntenatalRegister.xml]: Why is a concept of datatype N/A being used as a question of it self. Why not offer it as an answer to another question?

File: api/src/main/resources/metadata/Custom_Concepts.xml [hunk 3]
```diff
@@ -5265,6 +5346,92 @@
...
+    <concept_name  concept_name_id="161537"  concept_id="166200"  name="Relation PP/Brim"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-04 09:30:02"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="aa88f4fd-cf47-4a82-bf7a-ec424602a81a"  />
+    <concept_name  concept_name_id="161538"  concept_id="166201"  name="Foetal heart rate"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-04 09:31:46"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="27c287ce-f2a4-442b-bf25-411c0b426ce6"  />
+    <concept_name  concept_name_id="161539"  concept_id="166202"  name="pH"  locale="en"  locale_preferred="0"  creator="1"  date_created="2021-02-04 11:05:36"  concept_name_type="NULL"  voided="1"  voided_by="1"  date_voided="2021-02-04 11:13:52"  void_reason="Deleted due to change of name"  uuid="a26fc661-0896-4ad9-b45a-141f174ec167"  date_changed="2021-02-04 11:13:53"  changed_by="1"/>
```
Comment (slubwama) [inline api/src/main/resources/metadata/Custom_Concepts.xml:5392]: Why do you have a concept_name_type="NULL" you either leave it out or give it (FULLY_SPECIFIED,INDEX_TERM,SHORT).

File: api/src/main/resources/metadata/Custom_Concepts.xml [hunk 4]
```diff
@@ -5265,6 +5346,92 @@
...
+    <concept_name  concept_name_id="161497"  concept_id="166162"  name="Previous illness"  locale="en"  locale_preferred="0"  creator="1"  date_created="2021-02-02 10:41:55"  concept_name_type="NULL"  voided="1"  voided_by="1"  date_voided="2021-02-02 11:38:03"  void_reason="Deleted due to change of name"  uuid="b7ff2435-6619-425a-be3b-6a452a604f43"  date_changed="2021-02-02 11:38:03"  changed_by="1"/>
+    <concept_name  concept_name_id="161498"  concept_id="166162"  name="Previous Medical Illness"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-02 11:38:03"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="cb7b5f37-222c-40b6-a227-ead386a8d141"  date_changed="2021-02-02 11:38:03"  changed_by="1"/>
+    <concept_name  concept_name_id="161499"  concept_id="166163"  name="Operation"  locale="en"  locale_preferred="0"  creator="1"  date_created="2021-02-02 14:23:05"  concept_name_type="NULL"  voided="1"  voided_by="1"  date_voided="2021-02-02 15:05:45"  void_reason="Deleted due to change of name"  uuid="4e08d764-633c-4355-9f0e-4144c118586d"  date_changed="2021-02-02 15:05:45"  changed_by="1"/>
```
Comment (slubwama) [inline api/src/main/resources/metadata/Custom_Concepts.xml]: Why do you have a concept_name_type="NULL" you either leave it out or give it (FULLY_SPECIFIED,INDEX_TERM,SHORT).

File: api/src/main/resources/metadata/Custom_Concepts.xml [hunk 5]
```diff
@@ -5265,6 +5346,92 @@
...
     <concept_name  concept_name_id="161493"  concept_id="166158"  name="Missed Appointment Date"  locale="en"  locale_preferred="1"  creator="1"  date_created="2020-11-03 08:37:11"  concept_name_type="FULLY_SPECIFIED"  voided="0"   uuid="3aed47d1-a53a-4838-be7a-b51b7535a7f4"/>
 
+    <concept_name  concept_name_id="161496"  concept_id="166161"  name="Cardiac disease"  locale="en"  locale_preferred="1"  creator="1"  date_created="2021-02-01 16:45:34"  concept_name_type="FULLY_SPECIFIED"  voided="0" uuid="9c0c8516-c58e-4ee1-86df-48e66f921cc9"  />
+    <concept_name  concept_name_id="161497"  concept_id="166162"  name="Previous illness"  locale="en"  locale_preferred="0"  creator="1"  date_created="2021-02-02 10:41:55"  concept_name_type="NULL"  voided="1"  voided_by="1"  date_voided="2021-02-02 11:38:03"  void_reason="Deleted due to change of name"  uuid="b7ff2435-6619-425a-be3b-6a452a604f43"  date_changed="2021-02-02 11:38:03"  changed_by="1"/>
```
Comment (slubwama) [inline api/src/main/resources/metadata/Custom_Concepts.xml]: Why do you have a concept_name_type="NULL" you either leave it out or give it (FULLY_SPECIFIED,INDEX_TERM,SHORT).

**Review by slubwama — CHANGES_REQUESTED** [review]

**Comment by slubwama** [comment]
why-repeat-on-tests
Why on earth would you have a repeating section for tests

File: api/src/main/resources/metadata/Custom_Concepts.xml (hunk 5 above)
Comment (semujju) [reply api/src/main/resources/metadata/Custom_Concepts.xml]: This is resolved

File: api/src/main/resources/metadata/Custom_Concepts.xml (hunk 1 above)
Comment (semujju) [reply api/src/main/resources/metadata/Custom_Concepts.xml:5370]: This is resolved

> why-repeat-on-tests
> 
> Why on earth would you have a repeating section for tests
Why on earth 
ANC
would you design it that way

File: omod/src/main/webapp/resources/htmlforms/HMIS-005-IntegratedAntenatalRegister.xml (hunk 2 above)
Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/HMIS-005-IntegratedAntenatalRegister.xml]: I have instead created a new concept (for abortion)  which is boolean

File: api/src/main/resources/metadata/Custom_Concepts.xml (hunk 3 above)
Comment (semujju) [reply api/src/main/resources/metadata/Custom_Concepts.xml:5392]: This is resolved

File: api/src/main/resources/metadata/Custom_Concepts.xml (hunk 4 above)
Comment (semujju) [reply api/src/main/resources/metadata/Custom_Concepts.xml]: This is resolved

**Comment by Mwanje** [comment]
@semujju fix appointment return date it throws this "Cannot be in the future"

**Pull request merged**